### PR TITLE
tests: links: misc fixes

### LINKED ISSUE #1066: link checker: fix issues
- [ ] does not work on mac (sed is using some `-r` which is GNU-specific)
- [ ] does not print file names issue is found - https://circleci.com/gh/iterative/dvc.org/2592
- [ ] requires to be run from the root, it easy to make it run from any directory

### PR DESCRIPTION
**Pull request by casperdcl**
- [x] support Mac's version of `sed`
- [x] print file names of errors in diffs (useful in large PRs)
- [x] run from any dir (not just repo root)
- fixes #1066

### CODE REVIEW

File: scripts/link-check-git-all.sh [hunk 1]
```diff
@@ -1,3 +1,4 @@
 #!/usr/bin/env bash
-(find pages/ content/docs/ src/ .github/ -name '*.md' -o -name '*.js' && ls *.md *.js) \
+repo="$(dirname "$(realpath "$(dirname "$0")")")"
+(find "$repo"/pages/ "$repo"/content/docs/ "$repo"/src/ "$repo"/.github/ -name '*.md' -o -name '*.js' && ls "$repo"/*.md "$repo"/*.js) \
```
Comment (shcheklein) [inline scripts/link-check-git-all.sh]: does it fail if one of the directories does not exist?

Comment (casperdcl) [reply scripts/link-check-git-all.sh]: no - tested (exit status still zero). I presume you're talking about the warning

```
find: ‘...dvc.org/pages/’: No such file or directory
```

File: scripts/exclude-links.txt [hunk 2]
```diff
@@ -16,7 +16,6 @@ https://drive.google.com/drive/folders/0AIac4JZqHhKmUk9PDA
 https://dvc.org$
 https://dvc.org/doc/command-reference/foo
 https://dvc.org/foo
-https://dvc.org/img/<filename>.gif
```
Comment (casperdcl) [inline scripts/exclude-links.txt]: Should not have been required in #1025

File: scripts/link-check-git-all.sh (hunk 1 above)
Comment (shcheklein) [reply scripts/link-check-git-all.sh]: yes, would be good to catch changes in the docs structure if it's just a matter of adding some flag

File: scripts/exclude-links.txt (hunk 2 above)
Comment (shcheklein) [reply scripts/exclude-links.txt]: 👍

File: scripts/link-check-git-all.sh (hunk 1 above)
Comment (casperdcl) [reply scripts/link-check-git-all.sh]: I assume `pages` would be generated if the site was built. Are you saying it should be removed?

Comment (shcheklein) [reply scripts/link-check-git-all.sh]: no, no ... I'm not even sure if we need to test generated stuff .. just saying that next time someone decides to move `/content` to something like `/static/content` would be nice to detect that change ... again, not a big deal

Comment (casperdcl) [reply scripts/link-check-git-all.sh]: Right. Sounds like something for a different issue. Though it sounds like doing something like a directory move leaved the onus on the mover to update all tests.

Comment (shcheklein) [reply scripts/link-check-git-all.sh]: > like a directory move leaved the onus on the mover to update all tests

correct! but it's better for tests to start failing.

**Comment by shcheklein** [comment]
ready to merge? :)

**Pull request merged**